src/recipient-recieved-scanner.ts

```diff
@@ -292,4 +309,77 @@ export class RecipientRecievedScanner {
 
     return { startBlock, endBlock };
   }
+
+  /**
+   * Parses events and stores them using FileManager.
+   * @private
+   */
+  private async parseAndStoreEvents(
+    distributorAddress: string,
+    events: ethers.Log[],
+  ): Promise<void> {
+    // Load existing event data or initialize new structure
+    const existingData =
+      this.fileManager.readRecipientRecievedEvents(distributorAddress);
+
+    // Get chain ID from distributors data
+    const distributorsData = this.fileManager.readDistributors();
```

Comment:
We've called this once before - just pass the chain id around instead of calling the filemanager again